stage2/modlauncher9/src/main/java/gg/essential/loader/stage2/util/SortedJarOrPathList.java

```diff
@@ -52,6 +57,31 @@ private ArtifactVersion getVersion(Object pathOrJar) {
         if (metadata == null) return FALLBACK_VERSION;
         String version = metadata.version();
         if (version == null) return FALLBACK_VERSION;
+        // ModLauncher somehow manages to report the version of some jars (really unsure which ones, best guess
+        // right now is all those without a `FMLModType` manifest attribute? seemingly completely irregardless of
+        // whether they have an `Implementation-Version` attribute!)
+        // as "Optional.empty" (yes, that's a String, somewhere someone must have blindly `toString`ed).
```

Comment:
This was fixed [here](https://github.com/McModLauncher/securejarhandler/commit/07210811670ac3891565d665060c9458456b3a92#diff-37da836a20e7a2b48515e9a1bac7329cbbb2f1d1669b9d3a1dcd07386d8f919bL137), but not in forge's fork (not exactly sure what causes but, but definitely only affects modular jars, i.e. those with module-info.class). For present values Optional#toString does wrap the inner toString in `Optional[]`, does this affect anything?

Reply:
Hm, while I haven't ever observed a non-empty one (looking at the Java spec, there doesn't even appear to be a way to set your module's version, you have to manually tell Gradle to add it, so predicatively most people just don't have one set; only 270 search results in all of GitHub), we should definitely go unwrap those in case a future version of any of the libs we care about ends up doing it.
Now that I'm looking at the source of that bug, and the fact that it's been fixed in NeoForge, I realize that we should probably also remove the early `if (version == null) return FALLBACK_VERSION;` given there's a non-negligible chance that it might show up if they change something in their mod loading such that that class actually gets used for one of the libs we care about.